amber-slsa-buildtype.json

```diff
@@ -0,0 +1,137 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Amber SLSA buildType",
+  "description": "Describes a SLSA buildType, as defined in https://slsa.dev/provenance/v0.2.",
+  "type": "object",
+  "required": [
+    "buildType",
+    "buildConfig",
+    "materials"
+  ],
+  "not": {
+    "required": [
+      "invocation"
```

Comment:
Why is this not required?

Reply:
We don't have `invocation` in this schema. So I guess it somehow refers to the one in https://slsa.dev/provenance/v0.2. Is that right? 

The in-toto statements must specify a `predicateType`, which in case of SLSA provenances is:
```
"predicateType": "https://slsa.dev/provenance/v0.2",
```

With this schema, can we still use `https://slsa.dev/provenance/v0.2` or do we have to use this schema?

Reply:
You're correct that that's a reference to the SLSA schema. I'll add a comment to clarify that. We can still use their predicateType.

The reason this is omitted is that in SLSA, `invocation` and `buildConfig` hold the same information. We discussed this in the [Google document](https://docs.google.com/document/d/1xTEqPQxBG0iOSCbX1F79e1vx0zNelOoce00t6Z5-agU/edit?disco=AAAAUTPzORY) and agreed to use `buildConfig` only. :)

Reply:
Just to make sure I understand, this schema is not just a schema for the `buildType`. It is a **complete predicate specification** that conforms to the SLSA provenance spec (https://slsa.dev/provenance/v0.2#schema).

If so, I think it must include a `builder` too, since it is a [required field](https://slsa.dev/provenance/v0.2#fields) in the SLSA provenance.

Reply:
In my understanding this schema would merely lint that a SLSA file conforms to our build type. It does not actually check if it is a valid SLSA file. I went with this route so we wouldn't have to duplicate the SLSA schema definition, but happy to reconsider this approach.

Reply:
Do you have an example showing how the final in-toto statement would look like? The confusing part for me is that this schema mentions parts of the SLSA provenance (e.g., materials, buildConfig, and invocation) but not all of it. It is also unclear to me how we could use this schema for validating in-toto statements, with SLSA provenances and this build type.

Reply:
Can do! There's an example of how a provenance in-toto statement would look like in the in-toto repo: https://github.com/in-toto/attestation#provenance-example

I also created an example that uses our SLSA build type:

```
{
  "_type": "https://in-toto.io/Statement/v0.1",
  "subject": [
    {
      "name": "oak_functions_loader",
      "digest": {
        "sha256": "d4d5899a3868fbb6ae1856c3e55a32ce35913de3956d1973caccd37bd0174fa2"
      }
    }
  ],
  "predicateType": "https://slsa.dev/provenance/v0.2",
  "predicate": {
    "builder": {
      "id": "https://github.com/project-oak/transparent-release"
    },
    "buildType": "https://github.com/project-oak/transparent-release/amber-slsa-buildtype/v1.json",
    "buildConfig": {
      "command": "./scripts/runner build-functions-server",
      "outputPath": "./oak_functions/loader/bin/oak_functions_loader"
    },
    "materials": [
      {
        "uri": "gcr.io/oak-ci/oak@sha256:53ca44b5889e2265c3ae9e542d7097b7de12ea4c6a33785da8478c7333b9a320",
        "digest": {
          "sha256": "15dc16c42a4ac9ed77f337a4a3065a63e444c29c18c8cf69d6a6b4ae678dca5c"
        }
      },
      {
        "uri": "https://github.com/project-oak/oak",
        "digest": {
          "sha1": "0f2189703c57845e09d8ab89164a4041c0af0a62"
        }
      }
    ]
  }
}

```

If you'd like I could include an example file in this PR :)

Reply:
Update: added an example to the PR :)

Reply:
Also happy to hear your opinion if we should change the scope of the schema to validate that the statement is a valid SLSA file as well. 

My first intuition was no (to keep the scope small), but I realized that it would only require a few additional keys:
`_type`, `subject`, `predicate.builder.id`.

Reply:
Thanks. Let's add some code that validates the example against the schema (in a separate PR). We can add anything that is missing afterwards :)